client/chain/chain.go

```diff
@@ -833,7 +833,7 @@ func (c *chainClient) SyncBroadcastSignedTx(txBytes []byte) (*txtypes.BroadcastT
 	ctx := context.Background()
 
 	res, err := common.ExecuteCall(ctx, c.network.ChainCookieAssistant, c.txClient.BroadcastTx, &req)
-	if err != nil {
+	if err != nil || res.TxResponse.Code != 0 {
```

Comment:
In one of the examples that I drafted, an account  tries to send all the inj funds it posseses to itslef. It fails because it has no funds left to pay for gas fees. In such an example, `res.TxResponse.Code` code seems to be zero at this point of execution, so the function would not exit on `res.TxResponse.Code != 0` condition in such a case.